src/MCPServer/lib/server/queues.py

```diff
@@ -221,34 +197,35 @@ def stop(self):
         self.shutdown_event.set()
 
     def _package_completed_callback(self, package, future):
-        """Called by an Executor on completion the last job in a package.
+        """Called by an executor on completion of a package. Marks the package
+        as inactive and schedules a new package.
```

Comment:
One nit about your docstrings in this PR:

https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings
> Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description. The summary line may be used by automatic indexing tools; it is important that it fits on one line and is separated from the rest of the docstring by a blank line. The summary line may be on the same line as the opening quotes or on the next line. The entire docstring is indented the same as the quotes at its first line (see example below).

Reply:
Check, this was a bit careless of me. Fixed it!